sdk/resource/detect/env.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
Can this also be moved up one to the `resource` package? I forgot to include that in my previous comment, but it seems like calling `resource.FromEnv{}` is a decent enough name.